src/enhanced/utils/queue/index.ts

```diff
@@ -0,0 +1,48 @@
+import { SQS } from '@aws-sdk/client-sqs';
+import config from 'config';
+import { JobsQueuePayload } from '../../types/job-types';
+import { isJobQueuePayload } from '../../types/utils/type-guards';
+
+/**
+ * This function listens to the job queue until a message is received.
+ * @returns {Promise<JobsQueuePayload>} the promise for the payload object after a message has been received
+ */
+export async function listenToJobQueue(): Promise<JobsQueuePayload> {
+  const region = config.get<string>('aws_region');
+  const queueUrl = config.get<string>('jobsQueueUrl');
+
+  const client = new SQS({ region });
+
+  // We want to loop indefinitely so that we continue to poll the queue.
+  while (true) {
+    const receiveMessage = { QueueUrl: queueUrl, MaxNumberOfMessages: 1, WaitTimeSeconds: 2 };
+
+    const res = await client.receiveMessage(receiveMessage);
+
+    if (res.Messages && res.Messages.length > 0) {
+      const message = res.Messages[0];
+
+      if (!message.Body) {
+        console.error(`ERROR! Received message from queue without body. Message ID is: ${message.MessageId}`);
+        continue;
+      }
+
+      const payload = JSON.parse(message.Body);
+
+      // Use type guard here to validate payload we have received from the queue.
+      // This ensures that the `payload` object will be of type `JobQueuePayload` after the if statement.
+      if (!isJobQueuePayload(payload)) {
```

Comment:
Currently, we send messages through SQS with a minimal body ([class](https://github.com/mongodb/docs-worker-pool/blob/4feb61fef683a48c736422ff2977d5777688d93a/src/entities/queueMessage.ts#LL3C1-L3C1)), leveraging the job's ID for db queries as a source of truth.

Are we expecting to change the message's body in a future PR?

Reply:
Yes, we will change the payload moving forward. If my understanding is correct, we use the jobId to query the the queue collection in Atlas. Since we are no longer using that collection I am thinking we provide all of the data needed to complete the job as a part of the message.

Reply:
Gotcha. Passing in the data the Autobuilder would need to complete the job makes sense.

> Since we are no longer using that collection

Do you mean we're no longer using that collection for fetching jobs, or in general? I imagine we'd still be using the collection to save relevant information for auditing and debugging purposes

Reply:
Cassidy and I discussed, and we believe that the logging to CloudWatch should handle cover us in this regard. If you do think there is more value in keeping it around though, I'd be more than happy to discuss.

Reply:
Sure, that makes sense. I was mostly asking for my own understanding of all the changes coming through. I would argue that having a collection of jobs is still beneficial for having easy-to-see/query objects for reference, but if CloudWatch can achieve something similar, then no complaints.

I think the collection's only other uses are for sharing build logs to users, and for tracking metrics via our charts, but I think both of those can be achievable with CloudWatch/AWS magic.

Either way, this may not be relevant to the current PR so I'm down to pivot! Thanks for clearing things up for me and answering my questions!

Reply:
Gotcha! That all makes sense to me. CloudWatch should support these metrics, but you're right it might be more work, and I'm definitely flexible with how we go about this. Right now, we are trying to implement this without having to use the collection because it is a bit more difficult to test, and we are also looking to strip down the build process to the bare bones, and that requires not using that collection for now.

Anyhoo, we can definitely revisit this as this enhanced worker gets fleshed out more.